lib/puppet/parser/functions/uuid_generate.rb

```diff
@@ -0,0 +1,79 @@
+require 'digest/sha1'
+
+module Puppet::Parser::Functions
+  newfunction(:uuid_generate, :type => :rvalue, :doc => <<-END) do |args|
+
+    Creates a UUID based on a given string and an optional seed value.
+
+    For example, to generate a UUID based on the FQDN of a system:
+
+    Usage:
+
+      $uuid = uuid_generate($::fqdn)
+
+    The seed is optional. If given it must be a string.
+
+      $uuid_seeded = uuid_generate($::fqdn, 'seed-value')
+
+    The generated UUID will be the same for the given string and seed.
+
+    The resulting UUID is returned on the form:
+
+      1d839dea-5e10-5243-88eb-e66815bd7d5c
+
+    (u.e. without any curly braces.)
+
+    The generated UUID is a version 5 UUID with a custom namespace ID:
+
+      6ba7b810-9dad-11d1-80b4-00c04fd430c8
+
+    END
+
+    if args.length == 0
+      raise(ArgumentError, "uuid_generate: No arguments given")
+    elsif args.length == 1
+      seed = nil
+      fqdn = args[0]
+    elsif args.length == 2
+      fqdn = args[0]
```

Comment:
Technically it doesn't have to be an FQDN so it should be called `input_string` or something.